src/ed25519.rs

```diff
@@ -259,6 +260,9 @@ impl SignedPayload {
                 .try_into()
                 .map_err(|_| DecodeError::Invalid)?,
         );
+        if inner_payload_len > MAX_INNER_PAYLOAD_LENGTH {
```

Comment:
@C0x41lch0x41 I pushed a tweak in a2495af that has the len check here be against `64`, because actually the fn isn't supposed to support inner payloads greater than that length. At the top of the fn the entire payload is checked to fit inside a total length that requires the inner payload to be no longer than 64 bytes long.

Reply:
Yes that looks good to me thanks @leighmcculloch